### ISSUE TITLE
Event type of midiInput.onmidimessage wrong

### ISSUE THREAD

**Issue by h-a-n-n-e-s** [microsoft/TypeScript-DOM-lib-generator#1571]
This

`midiInput.onmidimessage = handleMIDIMessage;`
with
`handleMIDIMessage = (event:MIDIMessageEvent) => {...}`

runs perfectly fine in the browser, but the TypeScript compiler throws this error:

_Type '(event: MIDIMessageEvent) => void' is not assignable to type '(this: MIDIInput, ev: Event) => any'.
  Types of parameters 'event' and 'ev' are incompatible.
    Property 'data' is missing in type 'Event' but required in type 'MIDIMessageEvent'._

Shouldn't `midiInput.onmidimessage` have the type `(this: MIDIInput, ev: MIDIMessageEvent) => any` instead of `(this: MIDIInput, ev: Event) => any` ?

According to this
[https://developer.mozilla.org/en-US/docs/Web/API/MIDIMessageEvent](https://developer.mozilla.org/en-US/docs/Web/API/MIDIMessageEvent)
it should.

Thanks!

**Comment by christiaanwesterbeek** [comment]
I encountered this too. In my Chrome browser I can do this and it works.

```js
midiInput.onmidimessage = (msg) => {
    const [status, note, velocity] = msg.data;
    console.log(msg);
}
```

But still typescript complains that:

> Property 'data' does not exist on type 'Event'. ts(2339)

I changed  my code to the following and the typescript error is gone.

```js
midiInput.onmidimessage = (msg: MIDIMessageEvent) => {
    const [status, note, velocity] = msg.data;
    console.log(msg);
}
```

**Comment by h-a-n-n-e-s** [comment]
@christiaanwesterbeek
Yes, this is exactly what I do too, but Typescript erroneously expects type `Event` for `msg` and throws the error.

**Comment by fa-sharp** [comment]
Just submitted a PR to fix this :)

**Comment by h-a-n-n-e-s** [comment]
@fa-sharp 
Why can I not see your merge in my typescript 5.3.3 installation?
Shouldn't all updates here automatically be included in all typescript releases?

**Comment by fa-sharp** [comment]
@h-a-n-n-e-s Hmm, good question! I did assume it was automatic but maybe it isn't anymore.

I did a little digging, and I think the last time the DOM types from this library were merged into TypeScript was this PR: https://github.com/microsoft/TypeScript/pull/55798. So we might need to wait until TS 5.4 until the new types are merged? 🤷

**Comment by h-a-n-n-e-s** [comment]
@fa-sharp  Oh, I see. Then lets wait for the next minor release! ☺️ Thanks!

**Comment by h-a-n-n-e-s** [comment]
@fa-sharp In TS 5.4 it's merged! ☺️ Thanks again for your help!
